runtimes/eoapi/stac/eoapi/stac/client.py

```diff
@@ -90,4 +250,202 @@ async def landing_page(
+                request.headers.get("accept", ""), accepted_media
+            )
+
+        if output_type == MimeTypes.html:
+            return create_html_response(
+                request,
+                conforms_to,
+                template_name="conformance",
+            )
+
+        return conforms_to
+
+    async def all_collections(
+        self,
+        request: Request,
+        *args,
+        f: Optional[str] = None,
+        **kwargs,
+    ) -> Collections:
+        collections = await super().all_collections(request, *args, **kwargs)
+
+        output_type: Optional[MimeTypes]
+        if f:
+            output_type = MimeTypes[f]
+        else:
+            accepted_media = [MimeTypes[v] for v in get_args(ResponseType)]
+            output_type = accept_media_type(
+                request.headers.get("accept", ""), accepted_media
+            )
+
+        if output_type == MimeTypes.html:
+            return create_html_response(
+                request,
+                collections,
+                template_name="collections",
+                title="Collections list",
+            )
+
+        return collections
+
+    async def get_collection(
+        self,
+        collection_id: str,
+        request: Request,
+        *args,
+        f: Optional[str] = None,
+        **kwargs,
+    ) -> Collection:
+        collection = await super().get_collection(
+            collection_id, request, *args, **kwargs
+        )
+
+        output_type: Optional[MimeTypes]
+        if f:
+            output_type = MimeTypes[f]
+        else:
+            accepted_media = [MimeTypes[v] for v in get_args(ResponseType)]
+            output_type = accept_media_type(
+                request.headers.get("accept", ""), accepted_media
+            )
+
+        if output_type == MimeTypes.html:
+            return create_html_response(
+                request,
+                collection,
+                template_name="collection",
+                title=f"{collection_id} collection",
+            )
+
+        return collection
+
+    async def item_collection(
+        self,
+        collection_id: str,
+        request: Request,
+        *args,
+        f: Optional[str] = None,
+        **kwargs,
+    ) -> ItemCollection:
+        items = await super().item_collection(collection_id, request, *args, **kwargs)
+
+        output_type: Optional[MimeTypes]
+        if f:
+            output_type = MimeTypes[f]
+        else:
+            accepted_media = [MimeTypes[v] for v in get_args(GeoResponseType)]
+            output_type = accept_media_type(
+                request.headers.get("accept", ""), accepted_media
+            )
+
+        if output_type == MimeTypes.html:
+            items["id"] = collection_id
+            return create_html_response(
+                request,
+                items,
+                template_name="items",
+                title=f"{collection_id} items",
+            )
```

Comment:
yes it would be cool if we could add `elif output_type==MimeTypes.geoparquet`!

Reply:
yes 🙏